osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -249,20 +249,21 @@ public void SelectNextRandom()
 
             if (RandomAlgorithm == RandomSelectAlgorithm.RandomPermutation)
             {
-                var notYetVisitedSets = visible.Except(previouslyVisitedRandomSets).ToList();
+                var notYetVisitedSets = visibleSets.Except(previouslyVisitedRandomSets).ToList();
                 if (!notYetVisitedSets.Any())
                 {
-                    previouslyVisitedRandomSets.Clear();
-                    notYetVisitedSets = visible;
+                    previouslyVisitedRandomSets.RemoveAll(s => visibleSets.Contains(s));

```

Comment:
Does this really make a functional difference than just `Clear()`ing?

Reply:
Let's say you randomize through half of your maps then you wanna play only maps from some artist so you add a filter. Now you run out of maps while the filter is still on. Right now the code would also reset that 50% of other maps you were playing before even though you have more maps the randomizer hasn't selected yet if you didn't have a filter on. 

Just thought some users might experience that and then say that the randomizer is still not random enough.